### DISCUSSION TITLE
Error on build

### DISCUSSION THREAD

**Discussion by hi-reeve** [vuejs/pinia#783] [Help and Questions] [answered]
hello, i got an error while trying to build pinia.
this is my first time facing this error. 
any idea why is this?

```
node_modules/.pnpm/pinia@2.0.0-rc.10_typescript@4.4.4+vue@3.2.20/node_modules/pinia/dist/pinia.d.ts:128:138 - error TS2344: Type 'G' does not satisfy the constraint 'GettersTree<S>'.

128 export declare interface DefineStoreOptionsInPlugin<Id extends string, S extends StateTree, G, A> extends Omit<DefineStoreOptions<Id, S, G, A>, 'id' | 'actions'> {
```

nb: the error above with the pinia 2.0.0.rc-10

i tried upgrading pinia to the latest ```pinia 2.0.2```

and i got this error 

```
node_modules/.pnpm/pinia-plugin-persist@0.0.92_vue@3.2.20/node_modules/pinia-plugin-persist/dist/index.d.ts:1:21 - error TS2724: '"pinia"' has no exported member named 'GettersTree'. Did you mean '_GettersTree'?

1 import { StateTree, GettersTree, PiniaPluginContext } from 'pinia';
                      ~~~~~~~~~~~


Found 1 error.
```

@posva

**Comment by BenShelton** [comment, marked as answer]
Not sure about the issue with `rc-10`, does upgrading to the last release candidate (`rc-15`) fix the issue?

The issue when updating to `2.0.2` is because the plugin you're using `pinia-plugin-persist` hasn't been updated yet to work with the pinia 2 release. It's importing an internal type that has been deprecated in v2. You can:
1. Disable the type checking of node packages with `"skipLibCheck": true` in the `compilerOptions` of your tsconfig.json. This will ignore the type issue (see https://www.typescriptlang.org/tsconfig#skipLibCheck)
2. Even if the above works, raise an issue in that repo asking for full support for pinia 2: https://github.com/Seb-L/pinia-plugin-persist/issues

**Reply by hi-reeve** [reply]
removing the pinia plugin persist does solved the problem. thankyou!
